substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -3029,6 +3095,58 @@ impl<T: Config> Pallet<T> {
 		Ok(())
 	}
 
+	fn do_adjust_ed_deposit(who: T::AccountId, pool: PoolId) -> DispatchResult {
+		let bonded_pool = BondedPool::<T>::get(pool).ok_or(Error::<T>::PoolNotFound)?;
+		// only depositor can adjust ED deposit.
+		ensure!(bonded_pool.is_depositor(&who), Error::<T>::DoesNotHavePermission);
+
+		let reward_acc = &bonded_pool.reward_account();
+		let pre_frozen_balance =
+			T::Currency::balance_frozen(&FreezeReason::PoolMinBalance.into(), reward_acc);
+		let min_balance = T::Currency::minimum_balance();
+
+		if pre_frozen_balance == min_balance {
+			return Err(Error::<T>::NothingToAdjust.into())
+		}
+
+		// Update frozen amount with current ED.
+		Self::freeze_min_balance(reward_acc)?;
+
+		if pre_frozen_balance > min_balance {
+			// Transfer excess back to depositor.
+			let excess = pre_frozen_balance.saturating_sub(min_balance);
+			T::Currency::transfer(reward_acc, &who, excess, Preservation::Preserve)?;
+			Self::deposit_event(Event::<T>::MinBalanceExcessAdjusted {
+				pool_id: pool,
+				amount: excess,
+			});
+		} else {
+			// Transfer ED deficit from depositor to the pool
+			let deficit = min_balance.saturating_sub(pre_frozen_balance);
+			T::Currency::transfer(&who, reward_acc, deficit, Preservation::Expendable)?;
+			Self::deposit_event(Event::<T>::MinBalanceDeficitAdjusted {
+				pool_id: pool,
+				amount: deficit,
+			});
+		}
+
+		Ok(())
+	}
+
+	/// Apply freeze on reward account to restrict it from going below ED.
+	pub(crate) fn freeze_min_balance(reward_acc: &T::AccountId) -> DispatchResult {
```

Comment:
Similar to Kian's comment, this one-liner fn could be removed perhaps?

Reply:
It is one liner though it knows about what FreezeReason to use while freezing deposit which ideally only one fn should know about. We freeze deposit when pool is created and when pool deposit is adjusted and they should be frozen in exactly same way. I prefer this way (though its a minor thing and I am happy to change if you are not convinced) :)